### DISCUSSION TITLE
asyncValidate is triggered twice (onSubmit also)

### DISCUSSION THREAD

**Discussion by krasimirkostadinov** [react-hook-form/react-hook-form#6033] [General]
Hi, i have form with async validation for field, triggerred onBlur. Everything is ok, but onSubmit() this validation (async API call check is triggering again). Is there a way to skip this?

In my case i have multiple step form and on every step i submit subset of form fields. In this case when async check is triggerred again, it's delaying submit and this results to "flashing" screen before continue to next step (happening onSubmit callback). 

Is there way to skip running async validation check again onSubmit()  ?

**Comment by bluebill1049** [comment]
use `trigger` instead https://react-hook-form.com/api/useform/trigger
